src/comment_worker.py

```diff
@@ -269,25 +269,28 @@ def invest(self, sess, comment, investor, amount, suffix):
         """
         if config.PREVENT_INSIDERS:
             if comment.submission.author.name == comment.author.name:
-                comment.reply_wrap(message.INSIDE_TRADING_ORG)
+                return comment.reply_wrap(message.INSIDE_TRADING_ORG)
+
+        # Allows input such as '!invest 100%' and '!invest 50%'
+        if suffix == '%':
+            amount = int((investor.balance / 100) * int(amount))
+        else:
+            try:
+                amount = float(amount.replace(',', ''))
+                amount = int(amount * CommentWorker.multipliers.get(suffix, 1))
+            except ValueError:
                 return
 
-        try:
-            amount = float(amount.replace(',', ''))
-            amount = int(amount * CommentWorker.multipliers.get(suffix, 1))
-        except ValueError:
-            return
-
-        if amount < 100:
-            comment.reply_wrap(message.MIN_INVEST_ORG)
-            return
+        # Sets the minimum investment to 10% of an investor's balance
```

Comment:
You can move the percentage validation before the computations. You can check so 
``` python
if minim < 10 or minim > 100:
    return comment.reply_wrap(message.modify_invalid_percentage())
```

Assuming that you have a `message.modify_invalid_percentage()` that describes the percentage limits, so `10 <= percentage <= 100`

Reply:
The validation after the computation validates all three forms of investment (normal, abbreviated, and %), so adding more validation would lengthen the code. If you think we should do it we can, but as is it won't allow invalid amounts through.